internal/server/event_handlers.go

```diff
@@ -344,10 +339,19 @@ func extractAuthFromSecret(ctx context.Context, kubeClient client.Client, provid
 		options = append(options, notifier.WithHeaders(headers))
 	}
 
-	if user, ok := secret.Data["username"]; ok {
-		if pass, ok := secret.Data["password"]; ok {
-			options = append(options, notifier.WithUsername(strings.TrimSpace(string(user))))
-			options = append(options, notifier.WithPassword(strings.TrimSpace(string(pass))))
+	authMethods, err := secrets.AuthMethodsFromSecret(ctx, &secret)
+	if err == nil && authMethods != nil {
+		if authMethods.HasBearerAuth() {
+			options = append(options, notifier.WithToken(string(authMethods.Bearer)))
+		}
```

Comment:
This is a new API redundant with `token`, let's not add it

Reply:
OMG.
While I was working on this part, I realized I had forgotten to add support for KeyToken and submitted fluxcd/pkg#989, but I forgot to remove the Bearer Token code here 😢 
I'll delete it now.